nextcord/ext/commands/view.py

```diff
@@ -1,28 +1,188 @@
 # SPDX-License-Identifier: MIT
 
+from typing import Optional, Tuple, Type, overload
+
+from typing_extensions import Self
+
+import nextcord
+
 from .errors import ExpectedClosingQuoteError, InvalidEndOfQuotedStringError, UnexpectedQuoteError
 
-# map from opening quotes to closing quotes
-_quotes = {
-    '"': '"',
-    "‘": "’",
-    "‚": "‛",
-    "“": "”",
-    "„": "‟",
-    "⹂": "⹂",
-    "「": "」",
-    "『": "』",
-    "〝": "〞",
-    "﹁": "﹂",
-    "﹃": "﹄",
-    "＂": "＂",
-    "｢": "｣",
-    "«": "»",
-    "‹": "›",
-    "《": "》",
-    "〈": "〉",
-}
-_all_quotes = set(_quotes.keys()) | set(_quotes.values())
+__all__ = (
+    "Separator",
+    "Quotation",
+)
+
+
+class Separator:
+    """An argument delimiter.
+
+    Attributes
+    ----------
+    value: Optional[:class:`str`]
+        The value for the argument delimiter. By default, this is a space.
+    strip_ws: bool
+        Whether to strip surrounding whitespace
+    """
+
+    __slots__ = ("value", "strip_ws")
+
+    def __init__(self, value: Optional[str] = None, *, strip_ws: bool = True) -> None:
+        if value == "":
+            raise ValueError("Separators must be a non-empty string or None.")
+
+        self.value: str = value or " "
+        self.strip_ws: bool = strip_ws
+
+    def validate(self, character: str) -> bool:
+        """Validates a given character against a separator.
+
+        If the separator is any whitespace, then the character is checked to be whitespace
+        via :func:`str.isspace`.
+
+        Parameters
+        ----------
+        character: :class:`str`
+            The character to check this separator against.
+        """
+        if self.value.isspace():
+            return character.isspace()
+
+        return self.value == character
+
+    def __repr__(self) -> str:
+        return f"<Separator value={self.value!r} strip_ws={self.strip_ws}>"
+
+
+class Quotation:
+    """Delimiters for a quoted argument.
+
+    Parameters
+    ----------
+    start: :class:`str`
+        The character that starts a quotation.
+    end: Optional[:class:`str`]
+        The character that ends a quotation. By default, this is set to the start.
+    """
+
+    __slots__ = ("_values", "_is_default", "_cs_all_values", "_cs_initial_quotations")
+
+    @overload
+    def __init__(self, start: str, /) -> None:
+        ...
+
+    @overload
+    def __init__(self, start: str, end: str, /) -> None:
+        ...
+
+    def __init__(self, start: str, end: Optional[str] = None) -> None:
+        if start == "" or end == "":
+            raise ValueError("Quotations must be an non-empty string.")
+
+        self._values: dict[str, str] = {start: end or start}
```

Comment:
What version was `dict` typing introduced? Can't quite remember

Reply:
[3.9](https://peps.python.org/pep-0585/]

Reply:
Oh, I forgot the future annotations import.

Reply:
Any commits/changes in relation to this?

Reply:
I forgot about this earlier, sorry! Fixed in c67f468c.